examples/test_util/EclFile.cpp

```diff
@@ -0,0 +1,495 @@
+        case EIOD::CHAR  : return std::make_tuple(EIOD::sizeOfChar,EIOD::MaxBlockSizeChar);   break;
+        case EIOD::MESS  : throw std::invalid_argument("Type 'MESS' have not assosiated data") ;   break;
+    }
+}
+
+std::string trimr(const std::string &str1) {
+
+    std::string tmpStr=str1;
+    
+    while (tmpStr.back()==' '){
+        tmpStr.pop_back();
+    }
+
+    return std::move(tmpStr);
+}
+
+
+void readBinaryHeader(std::fstream &fileH,std::string &arrName,int &size,EIOD::eclArrType &arrType){
+
+    int bhead;
+    std::string tmpStrName(8,' ');
+    std::string tmpStrType(4,' ');
+
+    if (!fileH.is_open()){
+        std::string message="fstream fileH not open for reading";
+        throw std::runtime_error(message);
+    }
+
+    fileH.read((char *)&bhead, sizeof(bhead));
+    bhead=flipEndianInt(bhead);
+    
+    if (bhead!=16){
+        std::string message="Error reading binary header. Expected 16 bytes of header data, found " + std::to_string(bhead);
+        throw std::runtime_error(message);
+    }
+    
+    fileH.read(&tmpStrName[0], 8);
+    
+    fileH.read((char *)&size, sizeof(size));
+    size=flipEndianInt(size);
+
+    fileH.read(&tmpStrType[0], 4);
+
+    fileH.read((char *)&bhead, sizeof(bhead));
+    bhead=flipEndianInt(bhead);
+    
+    if (bhead!=16){
+        std::string message="Error reading binary header. Expected 16 bytes of header data, found " + std::to_string(bhead);
+        throw std::runtime_error(message);
+    }
+    
+    arrName=tmpStrName;
+    
+    if (tmpStrType=="INTE")
+        arrType=EIOD::INTE;
+    else if (tmpStrType=="REAL")    
+        arrType=EIOD::REAL;
+    else if (tmpStrType=="DOUB")    
+        arrType=EIOD::DOUB;
+    else if (tmpStrType=="CHAR")    
+        arrType=EIOD::CHAR;
+    else if (tmpStrType=="LOGI")    
+        arrType=EIOD::LOGI;
+    else
+        throw std::runtime_error("Error, unknown array type '" + tmpStrType +"'");
+    
+}
+
+
+unsigned long int sizeOnDisk(const int num, const EIOD::eclArrType &arrType){
+    
+    unsigned long int size=0;
+
+    if (arrType==EIOD::MESS){
+        if (num>0) {
+            std::string message="In routine calcSizeOfArray, type MESS can not have size > 0";
+            throw std::invalid_argument(message);
+        }
+    } else {
+
+        auto sizeData = block_size_data(arrType);
+
+	int sizeOfElement = std::get<0>(sizeData);
+        int maxBlockSize = std::get<1>(sizeData);
+        int maxNumberOfElements=maxBlockSize/sizeOfElement;
+       
+        size=num*sizeOfElement;
+        size=size + ((num-1) / maxNumberOfElements)*2*EIOD::sizeOfInte;    // 8 byte (two integers) every 1000 element        
+        
+        if (num>0){
+            size=size+2*EIOD::sizeOfInte;
+        }
+    }
+         
+    return size;
+}
+    
+
+
+
+std::vector<int> readBinaryInteArray(std::fstream &fileH,const int size){
+
+    std::vector<int> arr;
```

Comment:
This is in a reasonably tight loop - and we now the array size; it should be created with the correct size:
```
std::vector<int> arr(size);
```
and then `arr[index] = ...` should be used instead of `push_back()` in the inner loop.

Reply:
Added arr.reserve(size) which will allocate the needed capacity in single one allocation. push_back should then be ok.